lib/src/model.dart

```diff
@@ -3250,13 +3246,15 @@ abstract class ModelElement extends Canonicalization
 
   /// Override this to add more features to the documentation builder in a
   /// subclass.
-  String _buildDocumentationAddition(String docs) => docs;
+  String _buildDocumentationAddition(String docs) => docs ??= '';
 
   /// Separate from _buildDocumentationLocal for overriding.
   String _buildDocumentationBaseSync() {
+    // Prevent reentrancy.
     assert(_rawDocs == null);
```

Comment:
Instead of the comment, add an assert string:
```dart
assert(_rawDocs == null, '_buildDocumentationBaseSync cannot be re-entered');
```
Then it will be immediately obvious what the problem is when the assert fires.

Reply:
done.